### PR TITLE
Implement `RestStore.search_experiments`

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#6096

## What changes are proposed in this pull request?

- Depends on:
  - #6154
  - #6171 
- Implement `RestStore.search_experiments`.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/store/tracking/test_rest_store.py [hunk 1]
```diff
@@ -473,6 +474,40 @@ def test_databricks_paginate_list_experiments(self):
...
+            )
+
+        with mock_http_request() as mock_http:
+            store.search_experiments(order_by=["name DESC"])
+            self._verify_requests(
+                mock_http,
+                creds,
+                "experiments/search",
+                "POST",
+                message_to_json(SearchExperiments(view_type=ViewType.DELETED_ONLY)),
```
Comment (WeichenXu123) [inline tests/store/tracking/test_rest_store.py:508]: Can we add test to create a real request with backend handler routine run ?

Comment (harupy) [reply tests/store/tracking/test_rest_store.py]: Added :)

**Review by WeichenXu123 — APPROVED** [review]
LGTM, good work!

**Review by dbczumar — APPROVED** [review]
LGTM! @harupy before merging, can we add an integration test case to `test_rest_tracking`?

**Comment by harupy** [comment]
> LGTM! @harupy before merging, can we add an integration test case to `test_rest_tracking`?

@dbczumar Added :)

File: tests/tracking/test_rest_tracking.py [hunk 2]
```diff
@@ -759,19 +713,145 @@ def test_search_validation(mlflow_client):
...
+    assert [e.name for e in experiments] == reversed_experiment_names[:4]
+    assert experiments.token is not None
+
+    experiments = mlflow_client.search_experiments(max_results=4, page_token=experiments.token)
+    assert [e.name for e in experiments] == reversed_experiment_names[4:8]
+    assert experiments.token is not None
+
+    experiments = mlflow_client.search_experiments(max_results=4, page_token=experiments.token)
+    assert [e.name for e in experiments] == reversed_experiment_names[8:] + ["Default"]
+    assert experiments.token is None
```
Comment (harupy) [inline tests/tracking/test_rest_tracking.py:857]: This might be overkill.

Comment (harupy) [reply tests/tracking/test_rest_tracking.py]: Simplified the tests.

**Pull request merged**